packages/build/tests/plugins/fixtures/config_mutate_functions_bundling/plugin.js

```diff
@@ -0,0 +1,8 @@
+'use strict'
+
+module.exports = {
+  onPreBuild({ netlifyConfig }) {
+    // eslint-disable-next-line no-param-reassign
+    netlifyConfig.functions.test.node_bundler = 'zisi'
```

Comment:
When `node_bundler` is `esbuild`, the test snapshot is different because it prints a warning due to the usage of a dynamic require.

Reply:
We should avoid relying on this esbuild warning as the detection mechanism, because we're working on changing the way dynamic imports are handled. Could you use ES Modules instead, since ZISI will fail and esbuild will handle them correctly?

Reply:
It seems like `node_bundler: 'zisi'` is failing at runtime, not bundle time, with ES modules, so that would require unzipping the Functions archive.
Instead, I have changed it to adding a `require()` to an unknown dependency, which prints a different message with `precinct` than `esbuild`. Please let me know what you think.